1376 read images from extension

prupipho opened the pull request:
- Quick summary of changes
   - convert icons to base64 and show them from the templates folder
      - frameworks
      - pages...
- Which issue does this PR relate to?
   - #1376

prupipho committed:
- remove duplication for node.svg in client side
- some prettier format
- added svgBase64 &, iconBase64

prupipho committed:
first image used from templates folder

prupipho commented:
No need to review yet. Just want to see the build

prupipho committed:
- copy svg files on the templates folder
- remove all unnecessary items from client solution

prupipho committed:
- small tidy up
- TODO: create IconComponent
- TODO: update mock (mockImages rather than the real ones?)

prupipho committed:
create Icon component

prupipho committed:
- fix issue with icon appearing twice with if/else
- showing image for features/services from templates in ServicesPage
- remove unneeded svg and code to get those
- TODO: Missing still to show right image from rightsidebar

prupipho committed:
Merge remote-tracking branch 'upstream/dev' into prupipho-1376-readImagesFromExtension

prupipho commented on src/client/src/components/Icon/index.tsx:
```diff
@@ -0,0 +1,24 @@
+      ) : (
+        svgUrl && <img className={styles.icon} alt={name} src={svgUrl} />
+      )}
+      {/* //TODO Add missing image. Cancel SVG? duplicate and call default  */}
```
Remember

prupipho commented on src/client/src/mockData/extensionModules/mockData/mockWebPlatformData.ts:
```diff
@@ -103,6 +105,8 @@ const frameworks = [
     name: "Node",
     displayName: "Node.js/Express",
     icon: "",
+    iconBase64:
```
Do we want to put "mock" images rather than the ones that are currently on the templates?

prupipho commented on src/client/src/utils/cliTemplatesParser.ts:
```diff
@@ -12,14 +12,15 @@ export const getFrameworksOptions = (json: any[], type: FRAMEWORK_TYPE): IOption
     position: val.position,
     selected: val.selected,
-    svgUrl: "",
+    svgUrl: val.svgUrl, //maybe remove? as it´s always set to ""... or we leave that for the future... I vote to remove this
```
do we want to keep this here for the future? Or remove as it was being passed always as empty?

prupipho replied:
talked to Sibille and decided to remove that as we will always use the base64 image from the templates

prupipho commented on src/extension/src/client-modules/coreTSModule.ts:
```diff
@@ -21,8 +23,21 @@ export class CoreTSModule extends WizardServant {
+        item.iconBase64 = fs.readFileSync(item.icon, "base64");
+      });
+
+    //if some doesn´t have icon get default image
```
remember

prupipho committed:
check if this fix the build

prupipho committed:
- reorganised and removed unnecessary assets

prupipho committed:
send base64 from extension

prupipho committed:
icon component slightly modified
- small style
- empty image

prupipho committed:
show pages images by base64

prupipho committed:
updates to load page image

prupipho committed:
appservice and cosmosdb image updates

prupipho committed:
several updates to get framework images to work ok
removed style when no image. Looks weird and all of them should have img
missing files to commit for services changes to read images from templat

prupipho committed:
added mocks for svg images

prupipho committed:
Merge remote-tracking branch 'upstream/dev' into prupipho-1376-readImagesFromExtension

prupipho commented on src/client/src/pages/PageAddPages/PageCard/index.test.tsx:
```diff
@@ -26,46 +26,51 @@ describe("PageCard", () => {
   });
 
-  it("check img svg", ()=>{
+  xit("check img svg", () => {
```
is this type of test worth? shoul we think of trying to add testing for these?

prupipho committed:
remove comments

prupipho committed:
remove comments and update test

sibille commented:
Tested the PR, looks good to me!

There are a few unused svgs I think we can remove too:
- Azure.svg (and AzureIcon.tsx)
- AppService.svg
- CosmosDBWhite.svg
- FullStack.svg (and FullStack.tsx)

prupipho committed:
remove more unused svg,
remove unnecessary .tsx

prupipho committed:
few tests added to Icon component

prupipho committed:
small amend

Commit:
Merge branch 'dev' into prupipho-1376-readImagesFromExtension

Commit:
Remove svgUrl property

prupipho commented on src/client/src/types/apiTemplateInfo.d.ts:
```diff
@@ -5,7 +5,7 @@ export interface IApiTemplateInfo {
   longDescription: string;
   position: number;
-  svgUrl: string | undefined;
+  svgBase64?: string;
```
**svg**

prupipho commented on src/client/src/components/Icon/index.tsx:
```diff
@@ -0,0 +1,23 @@
+
+export interface IIconProps {
+  name: string;
+  svgBase64?: string;
```
name ICON => future iconType to allow png or others

prupipho replied:
Have a look at index,tsx VS Icon.tsx

prupipho replied:
No optional

prupipho commented on src/extension/src/client-modules/coreTSModule.ts:
```diff
@@ -21,8 +23,19 @@ export class CoreTSModule extends WizardServant {
     };
   }
 
+  private getBase64 = (array: any): any => {
```
rename **transformIconToBase64**

prupipho commented on src/extension/src/client-modules/coreTSModule.ts:
```diff
@@ -21,8 +23,19 @@ export class CoreTSModule extends WizardServant {
+    array
+      .filter((item: any) => item.icon !== null && item.icon !== "")
+      .forEach((item: any) => {
+        item.iconBase64 = fs.readFileSync(item.icon, "base64");
```
overwrite **ICON** value as this is when it communicates with client

prupipho commented on src/client/src/components/Icon/index.tsx:
```diff
@@ -0,0 +1,23 @@
+      {svgBase64 ? (
+        <img className={small ? styles.small : styles.icon} alt={name} src={"data:image/svg+xml;base64," + svgBase64} />
+      ) : (
+        <img className={styles.icon} />
```
ICON should be mandatory in the templates. Check how it´s shown in client and extension.

prupipho commented on src/client/src/mockData/extensionModules/mockData/mockSvgData.ts:
```diff
@@ -0,0 +1,11 @@
+const noImage = "";
```
imageToBase64 get from path and convert. One image for all.

prupipho committed:
cr changes:
- get rid of duplicated property with base64 name
- icon is a mandatory prop
- removed unneeded tests for no icon available

prupipho committed:
Merge branch 'dev' into prupipho-1376-readImagesFromExtension
